src/v/pandaproxy/json/requests/offset_commit.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "json/json.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/offset_commit.h"
+#include "kafka/protocol/offset_fetch.h"
+#include "kafka/protocol/schemata/offset_commit_request.h"
+#include "pandaproxy/json/requests/partition_offsets.h"
+#include "seastarx.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+namespace pandaproxy::json {
+
+inline std::vector<kafka::offset_commit_request_topic>
+partition_offsets_request_to_offset_commit_request(
+  std::vector<topic_partition_offset> tps) {
+    std::vector<kafka::offset_commit_request_topic> res;
+    if (tps.empty()) {
+        return res;
+    }
+
+    std::sort(tps.begin(), tps.end());
+    res.push_back(kafka::offset_commit_request_topic{tps.front().topic, {}});
```

Comment:
res.reserve(...)?

Reply:
> res.reserve(...)?

@dotnwat Ahh, this is the other `group_by`, let me see if I can make it an algorithm.